### ISSUE TITLE
[RFE]  Support configs in /usr/lib/ostree/ that get overridden by configs in /etc

### ISSUE THREAD

**Issue by sinnykumari** [ostreedev/ostree#1809]
While having a discussion on [Fedora CoreOS communit meeting](https://meetbot.fedoraproject.org/fedora-meeting-1/2019-01-30/fedora_coreos_meeting.2019-01-30-16.30.log.html), it came up that it will be a nice feature to support having configs in /usr/lib/ostree/ . Users can override these configs if needed by specifying it in /etc/ . This will allow us to debug and fix easily if any issue occurs on the user's machine is due to their local config changes.

**Comment by sinnykumari** [comment]
We can make use of this feature in https://github.com/coreos/fedora-coreos-tracker/issues/143 , where we are planning to ship ostree remote config in an rpm.

**Comment by sinnykumari** [comment]
> For the purposes of this ticket though, we're essentially talking about remote configs in system scenarios, right? IOW, I think we're talking about a `/usr/lib/ostree/remotes.d` dir with overrides in `/etc/ostree/remotes.d` 

Yes, we are talking here (https://github.com/coreos/fedora-coreos-tracker/issues/143) about having remote config in `/usr/lib/ostree/remotes.d/ dir with overrides in /etc/ostree/remotes.d/ . 

I made this RFE generic because I thought we might have other configs as well which we want to override.

> and possibly `/run/ostree/remotes.d`?

didn't know that this exist as well

**Comment by rfairley** [comment]
Related: #1801 to support `{/etc,/usr/lib}/ostree/bootloaders` (once `sysroot.bootloader` is a repo config key in #1814).
